system/lib/wasmfs/js_impl_backend.h

```diff
@@ -66,50 +68,51 @@ int _wasmfs_jsimpl_get_size(js_index_t backend, js_index_t index);
 namespace wasmfs {
 
 class JSImplFile : public DataFile {
-  js_index_t getBackendIndex() {
-    static_assert(sizeof(backend_t) == sizeof(js_index_t), "TODO: wasm64");
-    return js_index_t(getBackend());
-  }
+  static std::atomic<js_index_t> next_index;
```

Comment:
This worries me slightly. Backends are not often created so maybe a monotonic index is ok. But files might be created and destroyed constantly. Seems unlikely to overflow a 32-bit number, but maybe in a long-running process? At one file a second, that can happen in 136 years, which seems silly... but I don't know a reason why this couldn't happen 136 times a second, which would mean 1 year.

Perhaps we can just use the pointer value for both of them? It seems simpler, and just costs wasm64.